### PR TITLE
Added the code for v2 GET API settings/get/cadreConfig

### PR DESCRIPTION
**Pull request by ruksana2808**

### CODE REVIEW

File: service/src/main/java/org/sunbird/service/systemsettings/SystemSettingsService.java [hunk 1]
```diff
@@ -82,4 +89,44 @@ public <T> T getSystemSettingByFieldAndKey(
     }
     return null;
   }
+
+
+
+
+  public JsonNode getSystemSettingV2ByKey(String key, RequestContext context) {
+    logger.info("......................");
```
Comment (karthik-tarento) [inline service/src/main/java/org/sunbird/service/systemsettings/SystemSettingsService.java]: Remove unnecessary logs

File: service/src/main/java/org/sunbird/service/systemsettings/SystemSettingsService.java [hunk 2]
```diff
@@ -82,4 +89,44 @@ public <T> T getSystemSettingByFieldAndKey(
...
+    try {
+      DataCacheHandler.getConfigSettings().get(key);
```
Comment (karthik-tarento) [inline service/src/main/java/org/sunbird/service/systemsettings/SystemSettingsService.java]: This line seems unnecessary

File: service/src/main/java/org/sunbird/service/systemsettings/SystemSettingsService.java [hunk 3]
```diff
@@ -82,4 +89,44 @@ public <T> T getSystemSettingByFieldAndKey(
...
+//      setting = new SystemSetting(key, key, value);
+        ((ObjectNode) responseJson).put(JsonKey.ID, key);
+        ((ObjectNode) responseJson).put(JsonKey.FIELD, key);
+        ((ObjectNode) responseJson).put(JsonKey.VALUE, objectMapper.readTree(value));
+      } else {
+        setting = systemSettingDaoImpl.readByField(key, context);
+        ((ObjectNode) responseJson).put(JsonKey.ID, key);
+        ((ObjectNode) responseJson).put(JsonKey.FIELD, key);
+        ((ObjectNode) responseJson).put(JsonKey.VALUE, objectMapper.readTree(setting.getValue()));
+        if (null == setting) {
```
Comment (karthik-tarento) [inline service/src/main/java/org/sunbird/service/systemsettings/SystemSettingsService.java]: this suppose to be 1st line inside else case

File: service/src/main/java/org/sunbird/service/systemsettings/SystemSettingsService.java [hunk 4]
```diff
@@ -82,4 +89,44 @@ public <T> T getSystemSettingByFieldAndKey(
...
+
+
+
+
+  public JsonNode getSystemSettingV2ByKey(String key, RequestContext context) {
+    logger.info("......................");
+    try {
+      DataCacheHandler.getConfigSettings().get(key);
+      String value = DataCacheHandler.getConfigSettings().get(key);
+      SystemSetting setting = new SystemSetting();
```
Comment (karthik-tarento) [inline service/src/main/java/org/sunbird/service/systemsettings/SystemSettingsService.java]: this variable initialisation is unnecessary... remove this line

File: service/src/main/java/org/sunbird/service/systemsettings/SystemSettingsService.java [hunk 5]
```diff
@@ -82,4 +89,44 @@ public <T> T getSystemSettingByFieldAndKey(
...
+      JsonNode responseJson = objectMapper.createObjectNode();
+      if (value != null) {
+//      setting = new SystemSetting(key, key, value);
+        ((ObjectNode) responseJson).put(JsonKey.ID, key);
+        ((ObjectNode) responseJson).put(JsonKey.FIELD, key);
+        ((ObjectNode) responseJson).put(JsonKey.VALUE, objectMapper.readTree(value));
+      } else {
+        setting = systemSettingDaoImpl.readByField(key, context);
```
Comment (karthik-tarento) [inline service/src/main/java/org/sunbird/service/systemsettings/SystemSettingsService.java]: just declare the variable here

File: service/src/main/java/org/sunbird/service/systemsettings/SystemSettingsService.java [hunk 6]
```diff
@@ -82,4 +89,44 @@ public <T> T getSystemSettingByFieldAndKey(
...
+              ResponseCode.resourceNotFound.getErrorMessage(),
+              ResponseCode.RESOURCE_NOT_FOUND.getResponseCode());
+        }
+        DataCacheHandler.getConfigSettings().put(key, setting.getValue());
+      }
+      return responseJson;
+    }catch (Exception e){
+      logger.error(
+          context,
+          "Error occured during : "
```
Comment (karthik-tarento) [inline service/src/main/java/org/sunbird/service/systemsettings/SystemSettingsService.java]: Update the error message - mention "Failed to read system setting key: " and provide the key details which we are trying to read...

File: service/src/main/java/org/sunbird/actor/systemsettings/SystemSettingsActor.java [hunk 7]
```diff
@@ -52,4 +56,14 @@ private void setSystemSetting(Request actorMessage) {
...
     sender().tell(response, self());
   }
+
+  private void getSystemSettingV2(Request actorMessage) {
+    JsonNode setting =
+        service.getSystemSettingV2ByKey(
+            (String) actorMessage.getContext().get(JsonKey.FIELD),
+            actorMessage.getRequestContext());
+    Response response = new Response();
+    response.put(JsonKey.RESPONSE, setting);
```
Comment (karthik-tarento) [inline service/src/main/java/org/sunbird/actor/systemsettings/SystemSettingsActor.java]: there is a chance that null is returned for setting variable... in that case we return 200 OK with null value ??
check this out and make sure we return 404 error when the key is not found with proper json response body with proper errmsg

**Review by karthik-tarento — CHANGES_REQUESTED** [review]

File: service/src/main/java/org/sunbird/service/systemsettings/SystemSettingsService.java (hunk 1 above)
Comment (ruksana2808) [reply service/src/main/java/org/sunbird/service/systemsettings/SystemSettingsService.java]: done

File: service/src/main/java/org/sunbird/service/systemsettings/SystemSettingsService.java (hunk 3 above)
Comment (ruksana2808) [reply service/src/main/java/org/sunbird/service/systemsettings/SystemSettingsService.java]: done

**Pull request merged**